#Exercice 1 
#What is Data Analysis?


#answer : Data analysis can be defined as the process of collecting, transforming, studying data in order to extract meaningful and valuable insights that will support easy decision making 


#Question 2 : Why is Data analysis important in modern contexts?
nowadays, data analysis is crucial because of its ability to transform raw meaningless and messy data into strong insights that companies exploit
to explore differents business ideas and ensure high profit and better decisions making in their investments.

#Question 3 : List and describe three areas where data analysis is applied today.
1- Data analysis is strongly applied in sports today for example football, strong and detailed analytics give accurate prediction on team performance and outcome ahead of games.

2- Data analysis is also present in banking, specially when we talk about credit scoring, deciding to grant a customer a loan or not depending on his financial behaviour such as revenue, settled debt, time of settlement of past loans

3- Data analysis also helps in Healthcare, by conveying a diagnostic analysis on a patient the doctor is able to detect the rate of illness, prescribe cares in order to keep the patient in good fit

#Exercice 2 
the below DataSet describe the sleeping hours of a sample of US citizens aged from 15 years old per gender

In [1]:
import pandas as pd 

In [2]:
df = pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\Time Americans Spend Sleeping.csv")
print(df.head())

   index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both  


#The dataset imported below brings together an analysis of depression rate in Afghanistan and their possibles causes

In [3]:
df = pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\Mental health Depression disorder Data.csv")
print(df.head())

   index       Entity Code  Year Schizophrenia (%) Bipolar disorder (%)  \
0      0  Afghanistan  AFG  1990           0.16056             0.697779   
1      1  Afghanistan  AFG  1991          0.160312             0.697961   
2      2  Afghanistan  AFG  1992          0.160135             0.698107   
3      3  Afghanistan  AFG  1993          0.160037             0.698257   
4      4  Afghanistan  AFG  1994          0.160022             0.698469   

  Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
0             0.101855               4.828830                1.677082   
1             0.099313               4.829740                1.684746   
2             0.096692               4.831108                1.694334   
3             0.094336               4.830864                1.705320   
4             0.092439               4.829423                1.716069   

   Depression (%)  Alcohol use disorders (%)  
0        4.071831                   0.672404  
1        4.07953

C:\Users\DELL\AppData\Local\Temp\ipykernel_13052\599218142.py:1: DtypeWarning: Columns (0: Bipolar disorder (%), 1: Eating disorders (%)) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\Mental health Depression disorder Data.csv")


#the Dataset below is a credit scoring cleaned data set that gives details about customers race, marital status gender, rate of debt, gives them 1 if they are employed and 0 if they are not

In [4]:
df= pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\clean_dataset.csv")
print(df.head())

   Gender    Age   Debt  Married  BankCustomer     Industry Ethnicity  \
0       1  30.83  0.000        1             1  Industrials     White   
1       0  58.67  4.460        1             1    Materials     Black   
2       0  24.50  0.500        1             1    Materials     Black   
3       1  27.83  1.540        1             1  Industrials     White   
4       1  20.17  5.625        1             1  Industrials     White   

   YearsEmployed  PriorDefault  Employed  CreditScore  DriversLicense  \
0           1.25             1         1            1               0   
1           3.04             1         1            6               0   
2           1.50             1         0            0               0   
3           3.75             1         1            5               1   
4           1.71             1         0            0               0   

        Citizen  ZipCode  Income  Approved  
0       ByBirth      202       0         1  
1       ByBirth       43     560

In [5]:
df =pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\crx.csv")
print(df.head())

   b  30.83      0  u  g  w  v  1.25  t t.1  01  f g.1  00202  0.1  +
0  a  58.67  4.460  u  g  q  h  3.04  t   t   6  f   g  00043  560  +
1  a  24.50  0.500  u  g  q  h  1.50  t   f   0  f   g  00280  824  +
2  b  27.83  1.540  u  g  w  v  3.75  t   t   5  t   g  00100    3  +
3  b  20.17  5.625  u  g  w  v  1.71  t   f   0  f   s  00120    0  +
4  b  32.08  4.000  u  g  m  v  2.50  t   f   0  t   g  00360    0  +


#Exercice 4 
Here is the analysis of each dataset in text:

---

## 1. `clean_dataset.csv` — Credit Card Approval (690 rows, 16 columns)

**Gender** is **qualitative nominal** because even though it is encoded as 0/1, it represents a category (Male/Female) with no natural order.

**Age** is **quantitative continuous** because it is a numeric value with decimals representing a measurable age.

**Debt** is **quantitative continuous** because it represents an amount of debt — a measurable numeric value.

**Married** is **qualitative nominal** because the 0/1 encoding represents a Yes/No answer, which is a category.

**BankCustomer** is **qualitative nominal** because 0/1 represents whether someone belongs to a bank or not — a category.

**Industry** is **qualitative nominal** because it contains sector names (Industrials, Materials, etc.) with no natural order.

**Ethnicity** is **qualitative nominal** because it groups identities (White, Black, etc.) with no hierarchy.

**YearsEmployed** is **quantitative continuous** because it is a duration measured in years with decimals.

**PriorDefault** is **qualitative nominal** because 0/1 means "has previously defaulted or not" — a Yes/No category.

**Employed** is **qualitative nominal** because 0/1 indicates whether the person is employed or not — a category.

**CreditScore** is **quantitative discrete** because it is an ordered, measurable whole number score.

**DriversLicense** is **qualitative nominal** because 0/1 indicates whether someone holds a license or not — a category.

**Citizen** is **qualitative nominal** because it contains labels like "ByBirth" representing a status, with no numeric meaning.

**ZipCode** is **qualitative nominal** because even though it looks numeric, it codes a geographic location — averaging zip codes makes no sense.

**Income** is **quantitative continuous** because it is a measurable monetary value.

**Approved** is **qualitative nominal** because 0/1 represents the outcome of a decision (approved/rejected) — the categorical target variable.

---

## 2. `crx.csv` — Raw Credit Risk (689 rows, 16 columns)

This dataset is the unprocessed version of the previous one, with no explicit column headers. Columns are identified by position.

**Column 1** (values "a"/"b") is **qualitative nominal** because these are letter codes representing categories with no order.

**Column 2** (e.g. 30.83) is **quantitative continuous** because these are decimal values most likely corresponding to age.

**Column 3** (e.g. 0, 4.46) is **quantitative continuous** because these are decimal values most likely corresponding to debt.

**Columns 4, 5, 6, 7** (values "u", "g", "w", "v") are **qualitative nominal** because they are letter codes with no order or numeric meaning.

**Column 8** (e.g. 1.25) is **quantitative continuous** because these are decimal values most likely corresponding to years employed.

**Columns 9 and 10** (values "t"/"f") are **qualitative nominal** because "t" and "f" stand for True/False — boolean categories.

**Column 11** (integers) is **quantitative discrete** because these are whole numbers most likely corresponding to a credit score.

**Column 12** (values "t"/"f") is **qualitative nominal** for the same reason as columns 9 and 10.

**Column 13** (values "g") is **qualitative nominal** because it is a letter category code.

**Column 14** (e.g. 00202) is **qualitative nominal** because it is a zip code — a location identifier with no arithmetic meaning.

**Column 15** (integers) is **quantitative continuous** because these are numeric values most likely corresponding to income.

**Column 16** (values "+"/"-") is **qualitative nominal** because "+"/"-" represents Approved/Rejected — the target variable.

---

## 3. `Mental_health_Depression_disorder_Data.csv` (108,553 rows, 11 columns)

**index** is **quantitative discrete** because it is a whole number used as a row identifier, though it carries no real analytical meaning.

**Entity** is **qualitative nominal** because it is a country or region name — a label with no natural order.

**Code** is **qualitative nominal** because it is an ISO country code (e.g. AFG) — a label, not a numeric value.

**Year** is **quantitative discrete** because it is a calendar year — an ordered, measurable integer value.

**Schizophrenia (%)**, **Bipolar disorder (%)**, **Eating disorders (%)**, **Anxiety disorders (%)**, **Drug use disorders (%)**, **Depression (%)**, and **Alcohol use disorders (%)** are all **quantitative continuous** because they represent prevalence percentages — decimal values on which statistical calculations such as mean and standard deviation are fully meaningful.

---

## 4. `Time_Americans_Spend_Sleeping.csv` (945 rows, 9 columns)

**index** is **quantitative discrete** because it is a whole number row identifier with no analytical significance.

**Year** is **quantitative discrete** because it is a calendar year — an ordered integer value.

**Period** is **qualitative ordinal** because values like "Annual" represent categories that carry a potential order (e.g. monthly, quarterly, annual).

**Avg hrs per day sleeping** is **quantitative continuous** because it is the main measurement in the dataset — decimal hours on which all arithmetic operations are meaningful.

**Standard Error** is **quantitative continuous** because it is a statistical measure of precision expressed as a decimal value.

**Type of Days** is **qualitative nominal** because values like "All days" or "Weekdays" are category labels with no numeric meaning.

**Age Group** is **qualitative ordinal** because values like "15 years and over" are categories with an implied age-based order.

**Activity** is **qualitative nominal** because values like "Sleeping" are simple descriptive labels.

**Sex** is **qualitative nominal** because values like "Both", "Male", and "Female" represent gender categories with no natural ranking.

In [6]:
#Exercise 4


df = pd.read_csv(r"C:\Users\DELL\Downloads\Iris Dataset\Iris Dataset\Iris.csv")
print(df.head())

   Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa


#Exerice 4 
Quantitative Columns
SepalLengthCm is quantitative continuous because it measures the length of the sepal in centimeters. It takes decimal values (e.g. 5.1, 4.9) and arithmetic operations like computing the average or range are fully meaningful.
SepalWidthCm is quantitative continuous because it measures the width of the sepal in centimeters. Like sepal length, it is a decimal physical measurement on which all mathematical operations make sense.
PetalLengthCm is quantitative continuous because it measures the length of the petal in centimeters. It is a real-valued physical dimension that can be compared, averaged, and analyzed statistically.
PetalWidthCm is quantitative continuous because it measures the width of the petal in centimeters. It follows the same reasoning as the other measurement columns — a decimal numeric value with full arithmetic meaning.

Qualitative Columns
Id is qualitative nominal because even though it is stored as an integer, it is simply a row identifier assigned to each flower sample. It carries no measurable meaning — computing the average Id, for example, would be pointless.
Species is qualitative nominal because it contains the name of the iris flower species (e.g. Iris-setosa, Iris-versicolor, Iris-virginica). These are distinct category labels with no natural numeric order or ranking between them. 

In [7]:
df=pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\Time Americans Spend Sleeping.csv")
print(df.head())

   index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both  


we can use here columns like Type of days, Sex, age group, for a trend analysis because that could help identify wich gender and which age group sleeps the most

#Exercice 6
identifying structured and unstructured data
- A company’s financial reports stored in an Excel file is structured
- Photographs uploaded to a social media platform. is unstructured
- A collection of news articles on a website is unstructured
- Inventory data in a relational database is structured
- Recorded interviews from a market research study is unstructured 



#Exercice 7
## Convert Unstructured Data to Structured Data

### 1. Blog posts about travel experiences
- Method: Use text extraction + natural language processing.
- Steps:
  - Extract key fields: destination, dates, activities, accommodations, costs, ratings.
  - Apply NLP techniques: entity recognition for places, sentiment analysis for opinions, keyword extraction for themes.
  - Store results in a table or JSON schema like `{title, location, date, highlights, sentiment, tags}`.
- Reasoning: Blog posts are free-form text, so structuring requires identifying recurring travel attributes and mapping them to fields.

### 2. Audio recordings of customer service calls
- Method: Transcribe audio and then classify/extract.
- Steps:
  - Transcribe speech to text using ASR (automatic speech recognition).
  - Use NLP to identify speaker roles, issue categories, resolution status, sentiment, and key phrases.
  - Save as structured records like `{call_id, customer_name, issue_type, sentiment, duration, resolution}`.
- Reasoning: Audio itself is unstructured; transcription makes the content analyzable, then extraction creates consistent fields for reporting and analytics.

### 3. Handwritten notes from a brainstorming session
- Method: Digitize handwriting and organize by topic/action.
- Steps:
  - Scan and apply handwriting recognition (OCR for handwriting).
  - Segment notes into ideas, decisions, action items, questions.
  - Tag each item and store in a structured format like `{note_id, type, text, owner, priority}`.
- Reasoning: Handwritten notes are messy and spatially arranged; digitization plus categorization turns scattered thoughts into searchable records.

### 4. A video tutorial on cooking
- Method: Extract audio transcript and scene metadata.
- Steps:
  - Transcribe spoken instructions.
  - Detect steps, ingredients, timings, and equipment from text.
  - Optionally analyze video frames for visual cues or chapter segmentation.
  - Save as structured recipe steps: `{step_number, instruction, ingredients, duration}`.
- Reasoning: The video’s value lies in sequential instructions and ingredients, so breaking it into a step-by-step schema makes it reusable and searchable.


#Exercice 8

In [14]:
df = pd.read_csv(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\train.csv")
print(df.head(3))

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked  
0      0  A/5 21171   7.2500   NaN        S  
1      0   PC 17599  71.2833   C85        C  


#Exercice 9


import pandas as pd

# Create a simple dataframe
simple_df = pd.DataFrame({
    "name": ["Alice", "Bob", "Charlie"],
    "age": [25, 30, 22],
    "city": ["Paris", "Madrid", "Berlin"]
})

# This line is used to display the dataframe in the notebook
print(simple_df)

# Export to Excel
excel_path = r"c:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\Day_1\simple_dataframe.xlsx"
print(f"Exported Excel file to: {excel_path}")

# Export to JSON
json_path = r"c:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\Day_1\simple_dataframe.json"
simple_df.to_json(json_path, orient="records", indent=2)
print(f"Exported JSON file to: {json_path}")


#Exercice 10

In [18]:
df = pd.read_json(r"C:\Users\DELL\Desktop\DI-BOOTCAMP\Week_2\posts.json")
print(df.head(5))

   userId  id                                              title  \
0       1   1  sunt aut facere repellat provident occaecati e...   
1       1   2                                       qui est esse   
2       1   3  ea molestias quasi exercitationem repellat qui...   
3       1   4                               eum et est occaecati   
4       1   5                                 nesciunt quas odio   

                                                body  
0  quia et suscipit\nsuscipit recusandae consequu...  
1  est rerum tempore vitae\nsequi sint nihil repr...  
2  et iusto sed quo iure\nvoluptatem occaecati om...  
3  ullam et saepe reiciendis voluptatem adipisci\...  
4  repudiandae veniam quaerat sunt sed\nalias aut...  
